src/flint/fmpz_mod.jl

```diff
@@ -203,14 +203,19 @@ end
 ###############################################################################
 
 function ^(x::ZZModRingElem, y::Int)
-  R = parent(x)
   if y < 0
-    x = inv(x)
-    y = -y
+    z = inv(x)
+    x = pow!(x, z, -y)
+    swap!(inv!(z), x)
```

Comment:
```suggestion
    z = inv(x)
    z = pow!(z, z, -y)
```
Wouldn't this be simpler?